### ISSUE TITLE
Support for GitHub Actions

### ISSUE THREAD

**Issue by pascalberger** [cake-contrib/Cake.Issues.Recipe#148]
GitHub Actions supports [service messages](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message). We should use them if running under GitHub Actions and `ShouldReportIssuesToBuildServer` is enabled.

**Comment by AdmiringWorm** [comment]
Additionally, these service messages could be part of their own _group_ when reporting it on GitHub Actions.

for instance, grouped by `MSBuild`, `InspectCode` etc (similar to how the HTML reports are generated kind of).

This can be achieved with the workflow commands
```console
"::group::GroupName"
# Additional messages here
"::endgroup::GroupName"
```

Example of the above (_in powershell, but should in theory work in cake as well_) can be seen here: https://github.com/WormieCorp/Cake.Addin.Analyzer/blob/develop/build.ps1#L1-L21

_NOTE: Nested groups are not possible AFAIK_

**Comment by pascalberger** [comment]
Support for GitHub Actions has been implemented with #150. Support for grouping will follow in a seprate PR.

**Comment by pascalberger** [comment]
Grouping has been implemented with #153. Grouping only seems to have an effect in the build log output, not in the annotation list.
